internal/pkg/service/stream/api/service/sink.go

```diff
@@ -2,53 +2,220 @@ package service
 	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/api/gen/stream"
+	api "github.com/keboola/keboola-as-code/internal/pkg/service/stream/api/gen/stream"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/definition"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/definition/key"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/dependencies"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/model"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/statistics"
 )
 
-func (s *service) CreateSink(_ context.Context, d dependencies.SourceRequestScope, payload *stream.CreateSinkPayload) (res *stream.Task, err error) {
-	return nil, errors.NewNotImplementedError()
-}
+//nolint:dupl // CreateSource method is similar
+func (s *service) CreateSink(ctx context.Context, d dependencies.SourceRequestScope, payload *api.CreateSinkPayload) (*api.Task, error) {
+	// Create entity
+	sink, err := s.mapper.NewSinkEntity(d.SourceKey(), payload)
+	if err != nil {
+		return nil, err
+	}
+
+	// Quick check before the task
+	if err := s.sinkMustNotExist(ctx, sink.SinkKey); err != nil {
+		return nil, err
+	}
+
+	// Create sink in a task
+	t, err := s.startTask(ctx, taskConfig{
+		Type:      "create.sink",
+		Timeout:   5 * time.Minute,
+		ProjectID: d.ProjectID(),
+		ObjectKey: sink.SinkKey,
+		Operation: func(ctx context.Context, logger log.Logger) task.Result {
+			if err := s.definition.Sink().Create(&sink, s.clock.Now(), d.RequestUser(), "New sink.").Do(ctx).Err(); err == nil {
+				result := task.OkResult("Sink has been created successfully.")
+				result = s.mapper.WithTaskOutputs(result, sink.SinkKey)
+				return result
+			} else {
+				return task.ErrResult(err)
+			}
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
 
-func (s *service) GetSink(context.Context, dependencies.SinkRequestScope, *stream.GetSinkPayload) (res *stream.Sink, err error) {
-	return nil, errors.NewNotImplementedError()
+	return s.mapper.NewTaskResponse(t)
 }
 
-func (s *service) ListSinks(context.Context, dependencies.SourceRequestScope, *stream.ListSinksPayload) (res *stream.SinksList, err error) {
-	return nil, errors.NewNotImplementedError()
+func (s *service) GetSink(ctx context.Context, d dependencies.SinkRequestScope, _ *api.GetSinkPayload) (*api.Sink, error) {
+	if err := s.sinkMustExists(ctx, d.SinkKey()); err != nil {
+		return nil, err
+	}
+
+	sink, err := s.definition.Sink().Get(d.SinkKey()).Do(ctx).ResultOrErr()
+	if err != nil {
+		return nil, err
+	}
+
+	return s.mapper.NewSinkResponse(sink)
 }
 
-func (s *service) UpdateSink(context.Context, dependencies.SinkRequestScope, *stream.UpdateSinkPayload) (res *stream.Task, err error) {
-	return nil, errors.NewNotImplementedError()
+func (s *service) ListSinks(ctx context.Context, d dependencies.SourceRequestScope, payload *api.ListSinksPayload) (*api.SinksList, error) {
```

Comment:
`d dependencies.SinkRequestScope`?  I would expect that the source dependencies are for `ListSources` function.